src/Anonymization/Anonymizator.php

```diff
@@ -42,15 +43,17 @@ class Anonymizator implements LoggerAwareInterface
     ];
 
     private OutputInterface $output;
+    private readonly Context $defaultContext;
 
     public function __construct(
         private DatabaseSession $databaseSession,
         private AnonymizerRegistry $anonymizerRegistry,
         private AnonymizationConfig $anonymizationConfig,
-        private ?string $salt = null,
+        ?Context $defaultContext = null,
     ) {
         $this->logger = new NullLogger();
         $this->output = new NullOutput();
+        $this->defaultContext = $defaultContext ?? new Context();
     }
 
```

Comment:
I in favor of keeping things the most simple as we can. Here, can _we_ use the both options for the `$defaultContext` property ?

1. If we always give a `$defaultContext`, I think we should not accept it to be null.
2. If we never give a `$defaultContext`, then we should remove it from params and always prepopulate it in constructor.

I think we should not introduce behavior we don't need because it is more code to maintain, and it makes things more complicated to understand.